oraclepaas/resource_mysql_service_instance.go

```diff
@@ -0,0 +1,662 @@
+										Sensitive:    true,
+										ValidateFunc: validation.StringLenBetween(2, 32),
+									},
+									"em_password": {
+										Type:      schema.TypeString,
+										Optional:  true,
+										ForceNew:  true,
+										Sensitive: true,
+									},
+									"em_username": {
+										Type:      schema.TypeString,
+										Optional:  true,
+										ForceNew:  true,
+										Sensitive: true,
+									},
+									"em_port": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										ForceNew:     true,
+										Computed:     true,
+										ValidateFunc: validation.IntBetween(1, 65535),
+									},
+								},
+							},
+						},
+						// this comes from the service.
+						"connect_string": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"ip_address": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"public_ip_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"service_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+
+			"release_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"base_release_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"em_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+		}, // end declaration
+	} // end return
+}
+
+func resourceOraclePAASMySQLServiceInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+
+	log.Printf("[DEBUG] Resource state: %#v", d.State())
+	log.Print("[DEBUG] Creating mySQL service instance")
+
+	mySQLClient, err := getMySQLClient(meta)
+	if err != nil {
+		return err
+	}
+	client := mySQLClient.ServiceInstanceClient()
+
+	input := mysql.CreateServiceInstanceInput{}
+	input.ServiceParameters, err = getServiceParameters(d)
+	if err != nil {
+		log.Printf("[Error] : Error while extracting MySQL Service Instance information : %s", err)
+		return err
+	}
+
+	input.ComponentParameters, err = getComponentParameters(d)
+	if err != nil {
+		log.Printf("[Error] : Error while extracting MySQL component information from TF file. : %s", err)
+		return err
+	}
+
+	log.Printf("[DEBUG] : Testing Create :%v : ", &input)
+
+	_, err = client.CreateServiceInstance(&input)
+
+	if err != nil {
+		log.Printf("[Error] : Error while creating MySQL Service Instance : %v", err)
+		return err
+	}
+
+	d.SetId(input.ServiceParameters.ServiceName)
+	return resourceOraclePAASMySQLServiceInstanceUpdate(d, meta)
+}
+
+/**
+getServiceParameters gets the values from the terraform resource file, and updates the inputParameter
+with the respective values for calling the "Create"
+*/
+func getServiceParameters(d *schema.ResourceData) (mysql.ServiceParameters, error) {
```

Comment:
Let's swap this to `expandServiceParameters` to keep it line with the other expand functions

Reply:
Done